File: 0011.Container With Most Water.cs
var height = new int[] { 1, 8, 6, 2, 5, 4, 8, 3, 7 };
var s = new Solution();
var r = s.MaxArea(height);
Console.WriteLine(r);

public class Solution
{
    public int MaxArea(int[] height)
    {
        int start = 0;
        int end = height.Length-1;
        int maxArea = -1;
        while (start < end)
        {
            int area = (end- start) * (height[start] < height[end] ? height[start++] : height[end--]);
            maxArea = area > maxArea ? area : maxArea;
        }
        return maxArea;
    }
}

File: 0011.Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        int start = 0;
        int end = height.length - 1 ;
        while(start < end){
            max = Math.max(max, Math.min(height[end], height[start]) * (end-start));
            if(height[start] > height[end]){
                end--;
            }else{
                start++;
            }
        }
        return max;
    }
}
